include/multiverso/table_interface.h

```diff
@@ -30,25 +31,26 @@ class WorkerTable {
   int GetAsync(Blob keys, const GetOption* option = nullptr);
   int AddAsync(Blob keys, Blob values, const AddOption* option = nullptr);
 
-  void Wait(int id);
+  void Wait(integer_t id);
 
-  void Reset(int msg_id, int num_wait);
+  void Reset(integer_t msg_id, integer_t num_wait);
 
-  void Notify(int id);
+  void Notify(integer_t id);
 
   virtual int Partition(const std::vector<Blob>& kv,
-    std::unordered_map<int, std::vector<Blob> >* out) = 0;
+   MsgType partition_type,
+   std::unordered_map<int, std::vector<Blob> >* out) = 0;
 
   virtual void ProcessReplyGet(std::vector<Blob>&) = 0;
 
   // add user defined data structure
 private:
   std::string table_name_;
   // assuming there are at most 2^32 tables
-  int table_id_;
+  integer_t table_id_;
```

Comment:
I suggest use int. 2^32 is a really large number for table id. Only use the typedef type with specific situation.